### DISCUSSION TITLE
Change or disable http response

### DISCUSSION THREAD

**Discussion by EmilianoMorello** [arendst/Tasmota#20482] [Q&A]
Hi everyone. I have a device that sends http commands has a tasmota device. When the tasmota device returns the response it is interpreted as an error by the requesting device. is it possible to change or disable tasmota response after sending http command? Thanks for your help.

**Comment by sfromis** [comment]
No such option exist. The party sending a command is expected to be able to handle a response.

**Reply by EmilianoMorello** [reply]
Unfortunately the part that sends the command is a VoIP phone and I have no possibility to modify the software.  Can I customize the tasmota firmware?

**Reply by s-hadinger** [reply]
If your tasmota device is esp32 based, you could handle a specific path yourself and return whatever you want

**Reply by EmilianoMorello** [reply]
yes, it is based on Esp32.  how can I do it?

**Reply by sfromis** [reply]
You could create a small web server using the Tasmota integration with the Berry language
https://tasmota.github.io/docs/Berry/#webserver-module

**Reply by EmilianoMorello** [reply]
I think it's simpler to modify the tasmota source code in the part where the responses to http are defined but I don't know where this happens.  can you help me find it?

**Reply by s-hadinger** [reply]
Actually there's a very simple way in Berry:

```berry
import webserver

def my_page_handler()
  import webserver
  var ret = tasmota.cmd("Status 4")    # fake command, use whatever you want here
  # test if result is ok
  webserver.content_response("It's all good man")
  webserver.content_stop()
end

webserver.on("/my_page", my_page_handler, webserver.HTTP_GET)
```

Then looking at it with `curl`:
```
> curl http://192.168.x.x/my_page
*   Trying 192.168.x.x:80...
* Connected to 192.168.x.x (192.168.x.x) port 80 (#0)
> GET /my_page HTTP/1.1
> Host: 192.168.x.x
> User-Agent: curl/8.1.2
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/plain
< Server: Tasmota/13.3.0.2 (ESP32-D0WD)
< Cache-Control: no-cache, no-store, must-revalidate
< Pragma: no-cache
< Expires: -1
< Content-Length: 17
< Connection: close
<
* Closing connection 0
It's all good man%
```

**Reply by EmilianoMorello** [reply]
ok, I will try this solution.  Thank you

**Reply by sfromis** [reply]
If you want the command to be part of the request....
`var ret = tasmota.cmd(webserver.arg('cmnd'))`

(Of course, beware of the security implication of such an open door)

**Reply by s-hadinger** [reply]
Even better!

**Reply by EmilianoMorello** [reply]
I tried the berry solution. It seems to work well. How can I load the script permanently? If I turn off the device I have to recharge it.

**Reply by s-hadinger** [reply]
Put it in `autoexec.be`

**Reply by EmilianoMorello** [reply]
I created autoexec.be on the filesystem and specified load("my_file.be") but unfortunately it doesn't work. :(

**Reply by sfromis** [reply]
Well, not seeing any messages, or exactly what you did where leaves no room for informed guesses.

**Reply by EmilianoMorello** [reply]
if I restart berry with the brrestart command autoexec.be is reloaded and everything works. I believe that when I restart the device the connection is not immediately active and this creates a problem in autoexec.be which contains references to webserver.

**Reply by EmilianoMorello** [reply]
I solved it by using rule to launch the brrestart command. Thanks for your help.

**Reply by sfromis** [reply]
Such things is why I suggested you too look for messages. But using rules and the `BrRestart` command are quite a detour..

Simpler would be something like:
`tasmota.add_rule('System#Init', /->webserver.on("/my_page", my_page_handler, webserver.HTTP_GET))`
thereby delaying the `webserver.on` till Tasmota has been fully initialized. Also nice to have all of it in one file.
